Fix return value check:
https://github.com/iliaal/php_excel/commit/f6ddb5b0fe24a701dcbd95e932fb7b4a5ea7f7c4
Fix parameter order:
https://github.com/iliaal/php_excel/commit/9fa10ad2ec3b0772baecf26d8f0f1bebc1b22e08
Fix getVerPageBreak, getHorPageBreak:
https://github.com/iliaal/php_excel/commit/38cafd8be5bd578e2fbc2146bc5112cece399662
Fix build with libxl 3.5.4:
https://github.com/iliaal/php_excel/commit/0ae7d20d58107d87cf51b987f3707a17597bc45d
Change int to long:
https://github.com/iliaal/php_excel/commit/a8ce68eff5699501a059bd8173bd0ea1aa1fc747
https://github.com/iliaal/php_excel/commit/d3568f6bc954a77d940567c23045a23e8e31d078
--- excel.c.orig	2013-11-18 06:56:00.000000000 -0600
+++ excel.c	2014-03-19 14:24:37.000000000 -0500
@@ -53,7 +53,9 @@
 #if LIBXL_VERSION >= 0x03020000
 #define xlBookSetRefR1C1 xlBookSetRefR1C1A
 #define xlBookRefR1C1 xlBookRefR1C1A
+#endif
 
+#if LIBXL_VERSION >= 0x03020000 && LIBXL_VERSION < 0x03050401
 enum libXLPictureType {PICTURETYPE_PNG, PICTURETYPE_JPEG, PICTURETYPE_WMF, PICTURETYPE_DIB, PICTURETYPE_EMF, PICTURETYPE_PICT, PICTURETYPE_TIFF, PICTURETYPE_ERROR = 0xFF};
 #endif
 
@@ -1227,7 +1229,7 @@
 	BOOK_FROM_OBJECT(book, object);
 
 	xlBookSetActiveSheet(book, id);
-	RETURN_BOOL(id != xlBookActiveSheet(book));
+	RETURN_BOOL(id == xlBookActiveSheet(book));
 }
 /* }}} */
 
@@ -2043,7 +2045,7 @@
 #else
 			if (xlSheetIsDate(sheet, row, col)) {
 #endif
-				int dt = _php_excel_date_unpack(book, d);
+				long dt = _php_excel_date_unpack(book, d);
 				if (dt == -1) {
 					return 0;
 				} else {
@@ -3414,8 +3416,13 @@
 	long row, to_row, col, to_col;
 	char *name;
 	int name_len;
+#if LIBXL_VERSION >= 0x03050401
+	long scope_id = SCOPE_WORKBOOK;
 
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "sllll|l", &name, &name_len, &row, &to_row, &col, &to_col, &scope_id) == FAILURE) {
+#else
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "sllll", &name, &name_len, &row, &to_row, &col, &to_col) == FAILURE) {
+#endif
 		RETURN_FALSE;
 	}
 
@@ -3433,7 +3440,11 @@
 
 	SHEET_FROM_OBJECT(sheet, object);
 
+#if LIBXL_VERSION >= 0x03050401
+	RETURN_BOOL(xlSheetSetNamedRange(sheet, name, row, to_row, col, to_col, scope_id));
+#else
 	RETURN_BOOL(xlSheetSetNamedRange(sheet, name, row, to_row, col, to_col));
+#endif
 }
 /* }}} */
 
@@ -3445,8 +3456,13 @@
 	zval *object = getThis();
 	char *val;
 	int val_len;
+#if LIBXL_VERSION >= 0x03050401
+	long scope_id = SCOPE_WORKBOOK;
 
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s|", &val, &val_len, &scope_id) == FAILURE) {
+#else
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &val, &val_len) == FAILURE) {
+#endif
 		RETURN_FALSE;
 	}
 
@@ -3457,7 +3473,11 @@
 
 	SHEET_FROM_OBJECT(sheet, object);
 
+#if LIBXL_VERSION >= 0x03050401
+	RETURN_BOOL(xlSheetDelNamedRange(sheet, val, scope_id));
+#else
 	RETURN_BOOL(xlSheetDelNamedRange(sheet, val));
+#endif
 }
 /* }}} */
 
@@ -3616,18 +3636,31 @@
 	char *name;
 	int name_len;
 	int rf, rl, cf, cl;
+#if LIBXL_VERSION >= 0x03050401
+	long scope_id = SCOPE_WORKBOOK;
+	int hidden = 0;
 
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s|l", &name, &name_len, &scope_id) == FAILURE) {
+#else
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &name, &name_len) == FAILURE) {
+#endif
 		RETURN_FALSE;
 	}
 
 	SHEET_FROM_OBJECT(sheet, object);
+#if LIBXL_VERSION >= 0x03050401
+	if (xlSheetGetNamedRange(sheet, name, &rf, &rl, &cf, &cl, scope_id, &hidden)) {
+#else
 	if (xlSheetGetNamedRange(sheet, name, &rf, &rl, &cf, &cl)) {
+#endif
 		array_init(return_value);
 		add_assoc_long(return_value, "row_first", rf);
 		add_assoc_long(return_value, "row_last", rl);
-		add_assoc_long(return_value, "col_first", cl);
-		add_assoc_long(return_value, "col_last", cf);
+		add_assoc_long(return_value, "col_first", cf);
+		add_assoc_long(return_value, "col_last", cl);
+#if LIBXL_VERSION >= 0x03050401
+		add_assoc_bool(return_value, "hidden", hidden);
+#endif
 	} else {
 		RETURN_FALSE;
 	}
@@ -3641,18 +3674,29 @@
 	zval *object = getThis();
 	long index;
 	int rf, rl, cf, cl;
+#if LIBXL_VERSION >= 0x03050401
+	int hidden = 0, scope_id;
+#endif
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &index) == FAILURE) {
 		RETURN_FALSE;
 	}
 
 	SHEET_FROM_OBJECT(sheet, object);
+#if LIBXL_VERSION >= 0x03050401
+	if (xlSheetNamedRange(sheet, (int)index, &rf, &rl, &cf, &cl, &scope_id, &hidden)) {
+#else
 	if (xlSheetNamedRange(sheet, (int)index, &rf, &rl, &cf, &cl)) {
+#endif
 		array_init(return_value);
 		add_assoc_long(return_value, "row_first", rf);
 		add_assoc_long(return_value, "row_last", rl);
-		add_assoc_long(return_value, "col_first", cl);
-		add_assoc_long(return_value, "col_last", cf);
+		add_assoc_long(return_value, "col_first", cf);
+		add_assoc_long(return_value, "col_last", cl);
+#if LIBXL_VERSION >= 0x03050401
+		add_assoc_bool(return_value, "hidden", hidden);
+		add_assoc_long(return_value, "scope", scope_id);
+#endif
 	} else {
 		RETURN_FALSE;
 	}
@@ -3688,7 +3732,7 @@
 	}
 
 	SHEET_FROM_OBJECT(sheet, object);
-	RETURN_LONG(xlSheetGetVerPageBreakSize(sheet));
+	RETURN_LONG(xlSheetGetVerPageBreak(sheet, index));
 }
 /* }}} */
 
@@ -3721,7 +3765,7 @@
 	}
 
 	SHEET_FROM_OBJECT(sheet, object);
-	RETURN_LONG(xlSheetGetHorPageBreakSize(sheet));
+	RETURN_LONG(xlSheetGetHorPageBreak(sheet, index));
 }
 /* }}} */
 
@@ -4876,11 +4920,17 @@
 PHP_EXCEL_ARGINFO
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Sheet_getNamedRange, 0, 0, 1)
 	ZEND_ARG_INFO(0, name)
+#if LIBXL_VERSION >= 0x03050401
+	ZEND_ARG_INFO(0, scope_id)
+#endif
 ZEND_END_ARG_INFO()
 
 PHP_EXCEL_ARGINFO
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Sheet_getIndexRange, 0, 0, 1)
 	ZEND_ARG_INFO(0, index)
+#if LIBXL_VERSION >= 0x03050401
+	ZEND_ARG_INFO(0, scope_id)
+#endif
 ZEND_END_ARG_INFO()
 
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Sheet_namedRangeSize, 0, 0, 0)
@@ -5366,6 +5416,10 @@
 	REGISTER_EXCEL_CLASS_CONST_LONG(book, "PICTURETYPE_PICT", PICTURETYPE_PICT);
 	REGISTER_EXCEL_CLASS_CONST_LONG(book, "PICTURETYPE_TIFF", PICTURETYPE_TIFF);
 #endif
+#if LIBXL_VERSION >= 0x03050401
+	REGISTER_EXCEL_CLASS_CONST_LONG(book, "SCOPE_UNDEFINED", SCOPE_UNDEFINED);
+	REGISTER_EXCEL_CLASS_CONST_LONG(book, "SCOPE_WORKBOOK", SCOPE_WORKBOOK);
+#endif
 	return SUCCESS;
 }
 /* }}} */
--- tests/062.phpt.orig	2013-11-18 06:56:00.000000000 -0600
+++ tests/062.phpt	2014-03-19 14:28:12.000000000 -0500
@@ -26,7 +26,7 @@
 --EXPECT--
 bool(true)
 bool(true)
-array(4) {
+array(5) {
   ["row_first"]=>
   int(1)
   ["row_last"]=>
@@ -35,9 +35,11 @@
   int(10)
   ["col_last"]=>
   int(10)
+  ["hidden"]=>
+  bool(false)
 }
 bool(false)
-array(4) {
+array(6) {
   ["row_first"]=>
   int(20)
   ["row_last"]=>
@@ -46,6 +48,10 @@
   int(30)
   ["col_last"]=>
   int(30)
+  ["hidden"]=>
+  bool(false)
+  ["scope"]=>
+  int(-1)
 }
 bool(false)
 int(2)
--- tests/063.phpt.orig	2013-11-18 06:56:00.000000000 -0600
+++ tests/063.phpt	2014-03-19 14:26:23.000000000 -0500
@@ -37,8 +37,8 @@
 bool(true)
 int(2)
 int(2)
-int(2)
-int(2)
-int(2)
-int(2)
+int(10)
+int(10)
+int(-1)
+int(-1)
 OK
